brave/brave-core
# Brave News Source Suggestions

Resolves https://github.com/brave/brave-browser/issues/25563

## Limitations
This PR isn't a complete implementation of suggestions, but I think it's a good first pass, and provides sensible suggestions. I'm documenting a few of the limitations here:
- Suggestions are based only on:
    a) Whether a source has been visited
    b) How similar a source is to subscribed sources.
    c) How similar a source is to visited sources.
- Only the users current locale is used for suggestions. In future we should be able to incrementally improve this.

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

1. Enable the brave://flags/#brave-news-v2 flag
2. Subscribe to the `The Drive` publisher
3. Reload the new tab page
4. In the Brave News dialog there should be a suggestions row. It should include (if the similarity matrix hasn't been changed), `PopularMechanics`, `Forbes`, `Car and Driver`. Suggestions may (or may not) be slightly different
